src/units.cpp

```diff
@@ -346,51 +399,41 @@ void Units::addUnit(const std::string &reference)
 void Units::addUnit(StandardUnit standardUnit, const std::string &prefix, double exponent,
                     double multiplier, const std::string &id)
 {
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    addUnit(reference, prefix, exponent, multiplier, id);
+    addUnit(standardUnitToString.at(standardUnit), prefix, exponent, multiplier, id);
 }
 
 void Units::addUnit(StandardUnit standardUnit, Prefix prefix, double exponent,
                     double multiplier, const std::string &id)
 {
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    const std::string prefixString = prefixToString.find(prefix)->second;
-    addUnit(reference, prefixString, exponent, multiplier, id);
+    addUnit(standardUnitToString.at(standardUnit), prefixToString.at(prefix), exponent, multiplier, id);
 }
 
 void Units::addUnit(StandardUnit standardUnit, int prefix, double exponent,
                     double multiplier, const std::string &id)
 {
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    const std::string prefixString = convertToString(prefix);
-    addUnit(reference, prefixString, exponent, multiplier, id);
+    addUnit(standardUnitToString.at(standardUnit), convertToString(prefix), exponent, multiplier, id);
 }
 
 void Units::addUnit(StandardUnit standardUnit, double exponent, const std::string &id)
 {
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    addUnit(reference, "0", exponent, 1.0, id);
+    addUnit(standardUnitToString.at(standardUnit), "0", exponent, 1.0, id);
 }
 
 void Units::addUnit(StandardUnit standardUnit)
 {
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    addUnit(reference, "0", 1.0, 1.0, "");
+    addUnit(standardUnitToString.at(standardUnit), "0", 1.0, 1.0, "");
 }
 
 void Units::unitAttributes(StandardUnit standardUnit, std::string &prefix, double &exponent, double &multiplier, std::string &id) const
 {
     std::string dummyReference;
-    const std::string reference = standardUnitToString.find(standardUnit)->second;
-    auto result = pFunc()->findUnit(reference);
-    unitAttributes(size_t(result - pFunc()->mUnitDefinitions.begin()), dummyReference, prefix, exponent, multiplier, id);
+    unitAttributes(size_t(pFunc()->findUnit(standardUnitToString.at(standardUnit)) - pFunc()->mUnitDefinitions.begin()), dummyReference, prefix, exponent, multiplier, id);
```

Comment:
Should probably be written as an explicit cast such as: `static_cast<sizet>()`.

Reply:
80e32368